src/main/java/interface_adapters/Controller.java

```diff
@@ -0,0 +1,97 @@
+package interface_adapters;
+
+import game_data.GameInstance;
+import usecase_gamedata.MapFactory;
+import usecase_gamedata.PlayerFactory;
+import usecase_playeractions.ActionManager;
+import usecase_playeractions.Map;
+
+import java.awt.event.KeyEvent;
+import java.awt.event.KeyListener;
+
+/**
+ * The Controller used to control player.
+ */
+public class Controller implements KeyListener {
+
+    /**
+     * Contain actions that depend on a Map.
+     * Don't have to be a Mover.
+     */
+    private ActionManager moveManager;
+
+    /**
+     * Contain actions that depend on a Player only.
+     */
+    private ActionManager manager;
+    private PlayerFactory playerFactory;
+
+
+    /**
+     * The default keys:
+     * 0:Heal
+     * 1:Heal-yes
+     * 2:Heal-No
+     * 3:Upgrade
+     * 4:Upgrade-Armor
+     * 5:Upgrade-Sword
+     * 6:Fight
+     * 7:Flee
+     * 8,9,10,11:moveUP,moveLeft,moveDown,moveRight
+     */
+    private final String[] KEYS = new String[]{"H","Y","N","U","A","S","F","R","W","A","S","D"};
```

Comment:
We will need to decide on a key for "Continue" because I have use cases which are triggered (ex: GameOver use case) after the user presses a continue button after a Monster Fight. Let's discuss in the discord